File: Ccode/Number Theroy/Magic Mod/1.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
const int mod = 1e9 + 7;

ll qpow(ll a, int b, int mod){
    ll res = 1;
   while(b > 0){
       if(b & 1) res = res * a % mod;
       a = a * a % mod;
       b >>= 1;
   }
   return res;
}

const int inv6 = qpow(6, mod - 2, mod);

inline ll f2(ll n){
    return n * (n + 1) % mod * (2 * n + 1) % mod * inv6 % mod;
}

ll calc(ll n){
    ll ans = n * (n + 1) / 2LL % mod * n % mod, res = 0;
    for(ll l = 1, r; l <= n; l = r + 1){
        r = n / (n / l);
        res = (res + (f2(r) - f2(l - 1) % mod + mod) % mod * (n / l) % mod) % mod;
    }
    ans = ((ans - res) % mod + mod) % mod;
    return ans;
}

int main(){
    ll n, m;
    while(cin >> n >> m){
        cout << calc(n) * calc(m) % mod << endl;
    }
    return 0;
}

File: Ccode/Number Theroy/Magic Mod/1.py
# def Min(a, b):
#     if a > b:
#         return b
#     else:
#         return a

# def calc(x, p):
#     res = 0
#     while x > 0:
#         res = res + x // p
#         x = x // p
#     return res


# prime = [2, 3, 5, 7, 11]
# a, b = map(int, input().split())
# ans = 1
# for v in prime:
#     ans = ans * (v ** Min(calc(a, v), calc(b, v)))

# print(ans) 

# s = input()
# print(max(map(int, (s, s[:-1], s[:-2] + s[-1]))))



n = 210
res = 0
cnt = 2
while cnt <= 20:
    res = res + n
    s = str(res)
    if len(s) > cnt:
        print(cnt + 1, res)
        cnt = cnt + 1
        

